File: source/livana-fe/src/types/response/reviewResponse.ts
export interface ReviewResponse {
  id: string;
  listingId: string;
  reviewerId: string;
  reviewerName: string;
  reviewerAvatar: string | null;
  rating: number;
  comment: string | null;
  reviewType: "HOME_LISTING" | "EXPERIENCE_LISTING";
  createdAt: string;
}

export interface ListingRatingSummary {
  averageRating: number;
  totalReviews: number;
  reviews: ReviewResponse[];
}

File: source/livana-be/src/main/java/octguy/livanabe/dto/response/ReviewResponse.java
package octguy.livanabe.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReviewResponse {
    
    private UUID id;
    private UUID listingId;
    private UUID reviewerId;
    private String reviewerName;
    private String reviewerAvatar;
    private Integer rating;
    private String comment;
    private String reviewType;
    private LocalDateTime createdAt;
}
